### ISSUE TITLE
Trimming and adding slits

### ISSUE THREAD

**Issue by kbwestfall** [pypeit/PypeIt#450]
In testing the 830G_M DEIMOS reduction in the dev suite, I think it actually removes 2 of the slits that it should find based on the slit mask because of the fracignore value.  I.e., I get the message:
```
[INFO]    :: traceslits.py 824 _trim_slits() - Slit 9 is less than fracignore - ignoring this slit
```
This led me to a few questions.  The description of `fracignore` is:
```
>>> par['calibrations']['slits'].descr['fracignore']
"If a slit spans less than this fraction over the spectral size of the detector, 
it will be ignored (and reconstructed when/if an 'order' PCA analysis is performed)."
```
 - Based on `traceslits.TraceSlits._trim_slits`, I think you mean "spatial", not "spectral", right?
 - Is the last part about the slit being reconstructed true?  `TraceSlits._pca` is only called before `TraceSlits._trim_slits` and not again after.
 - I added a `usefracpix` keyword to `TraceSlits.add_user_slits` that gets passed to `TraceSlits._trim_slits` when `run_to_finish=True`.  Otherwise it was just rejecting the slits again.

Having said all that, I'm getting some weird behavior with adding slits.  I've been testing using the notebook [here](https://github.com/pypeit/PypeIt/blob/add_mask_slits/doc/nb/CalibrationsClass_deimos.ipynb).

The bottom line is that I'm trying to add 3 slits.  It adds the first one successfully.  The second one is added, but it doesn't match the trace image well and it messes up one of the existing slits.  If I add all three things really go haywire: all the left traces are on the left side of the detector and all the right traces are on the right side of the detector.  I'll wade into the deep end on this next week, but let me know if you have a look at the notebook and ginga plots and think you might know what's going on.

**Comment by profxj** [comment]
This should have been addressed by the tracing slits refactor.
